lib/src/model/coursetable/course.dart

```diff
@@ -0,0 +1,101 @@
+import 'package:flutter_app/src/model/coursetable/course_period.dart';
+import 'package:json_annotation/json_annotation.dart';
+
+import '../json_init.dart';
+
+part 'course.g.dart';
+
+@JsonSerializable()
+class Course {
+  String idString;
+  late int id;
+
+  String name;
+
+  String stageString;
+  late double stage;
+
+  String creditString;
+  late double credit;
+
+  String periodCountString;
+  late int periodCount;
+
+  String category;
+
+  String teacherString;
+  late List<String> teachers;
+
+  String classNameString;
+  late List<String> classNames;
+
+  List<String> periodSlots;
+  late List<CoursePeriod> coursePeriods;
+
+  String classroomString;
+  late List<String> classrooms;
+
+  String applyStatus;
+  String language;
+  String syllabusLink;
+  String note;
+
+  Course(
+      {required this.idString,
+      required this.name,
+      required this.stageString,
+      required this.creditString,
+      required this.periodCountString,
+      required this.category,
+      required this.teacherString,
+      required this.classNameString,
+      required this.periodSlots,
+      required this.classroomString,
+      required this.applyStatus,
+      required this.language,
+      required this.syllabusLink,
+      required this.note}) {
```

Comment:
```suggestion
  Course({
    required this.idString,
    required this.name,
    required this.stageString,
    required this.creditString,
    required this.periodCountString,
    required this.category,
    required this.teacherString,
    required this.classNameString,
    required this.periodSlots,
    required this.classroomString,
    required this.applyStatus,
    required this.language,
    required this.syllabusLink,
    required this.note,
  }) {
```

Suggest always add a comma `,` after the final things in a series of long parameters.
Based on Flutter's rule: https://docs.flutter.dev/tools/formatting#using-trailing-commas

Reply:
Fix on b4b19a33f1e9bff451f913fcd014cfa04748ef2a